src/file.cpp

```diff
@@ -2036,58 +2006,12 @@ typedef struct _FILE_ALLOCATED_RANGE_BUFFER {
 			}
 		}
 
-		if ((m_open_mode & sparse) == 0)
-		{
-#if TORRENT_USE_WSTRING
-			typedef DWORD (WINAPI *GetCompressedFileSize_t)(LPCWSTR lpFileName, LPDWORD lpFileSizeHigh);
-#else
-			typedef DWORD (WINAPI *GetCompressedFileSize_t)(LPCSTR lpFileName, LPDWORD lpFileSizeHigh);
-#endif
-
-			static GetCompressedFileSize_t GetCompressedFileSize_ = NULL;
+		//if ((m_open_mode & sparse) == 0) {...}
+		// NOTE: previously, there was an attempt here to fill with valid data
+		// the sparse file, but the function SetFileValidData can't be used here
+		// by MSDN documentation.
+		// See https://msdn.microsoft.com/en-us/library/windows/desktop/aa365544(v=vs.85).aspx
```

Comment:
right. (open_mode & sparse) == 0 is true when the sparse flag is _not_ set. i.e. it's a regular fully allocated file